ambry-replication/src/main/java/com/github/ambry/replication/CloudToStoreReplicationManager.java

```diff
@@ -300,17 +300,22 @@ public void onInstanceConfigChange(List<InstanceConfig> instanceConfigs, Notific
         if (instanceConfig.getRecord().getBooleanField(VCR_HELIX_CONFIG_READY, false)) {
           // only when VCR_HELIX_CONFIG_READY, we take action on it.
           String instanceName = instanceConfig.getInstanceName();
-          Port sslPort =
-              getSslPortStr(instanceConfig) == null ? null : new Port(getSslPortStr(instanceConfig), PortType.SSL);
-          Port http2Port = getHttp2PortStr(instanceConfig) == null ? null
-              : new Port(getHttp2PortStr(instanceConfig), PortType.HTTP2);
-          CloudDataNode cloudDataNode = new CloudDataNode(instanceConfig.getHostName(),
-              new Port(Integer.parseInt(instanceConfig.getPort()), PortType.PLAINTEXT), sslPort, http2Port,
-              clusterMapConfig.clustermapVcrDatacenterName, clusterMapConfig);
-          newInstanceNameToCloudDataNode.put(instanceName, cloudDataNode);
-          newVcrNodes.add(cloudDataNode);
-          logger.info("Instance config change. VCR Node {} added. SslPort: {}, Http2Port: {}", cloudDataNode, sslPort,
-              http2Port);
+          if (!newInstanceNameToCloudDataNode.containsKey(instanceName)) {
```

Comment:
Are we assuming that an instance config change notification from helix will contain multiple notifications for the same instance?